modules/rntuple.mjs

```diff
@@ -92,41 +92,60 @@ class RBufferReader {
     this.offset += 8;
     return val;
   }
-  
+
 }
 
 
 class RNTupleDescriptorBuilder {
-      
-   deserializeHeader(header_blob) {
-    if (!header_blob) return;
-
-    const reader = new RBufferReader(header_blob);
 
-    // 1. Read header version (4 bytes)
-    this.version = reader.readU32();
+deserializeHeader(header_blob) {
+  if (!header_blob) return;
 
-    // 2. Read feature flags (4 bytes)
-    this.headerFeatureFlags = reader.readU32();
+  const reader = new RBufferReader(header_blob),
+        packed = reader.readU64();
 
-    // 3. Read xxhash3 (64-bit, 8 bytes)
-    this.xxhash3 = reader.readU64();
+  // Step 0: Envelope metadata
+  this.envelopeType = Number(packed & 0xFFFFn);
+  this.envelopeLength = Number((packed >> 16n) & 0xFFFFFFFFFFFFn);
```

Comment:
It's ok to do it later, but you should eventually parse the checksum of the envelope and validate it. You can add a TODO comment if you don't want to tackle it now.